File: src/main/java/com/prototype/architecture/layer_03_domain/auth/AuthUser.java
package com.prototype.architecture.layer_03_domain.auth;

import com.prototype.architecture.layer_03_domain.user.UserId;
import com.prototype.architecture.layer_03_domain.userrole.UserRoleType;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.GrantedAuthority;

import java.util.Collection;
import java.util.List;

public record AuthUser(
        UserId userId,
        List<UserRoleType> userRoleTypes
) implements Authentication {

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return userRoleTypes;
    }

    @Override
    public Object getCredentials() {
        return null;
    }

    @Override
    public Object getDetails() {
        return null;
    }

    @Override
    public Object getPrincipal() {
        return userId;
    }

    @Override
    public boolean isAuthenticated() {
        return true;
    }

    @Override
    public void setAuthenticated(boolean isAuthenticated) throws IllegalArgumentException {

    }

    @Override
    public String getName() {
        return null;
    }
}

File: src/main/kotlin/com/prototype/architecture/layer_03_domain/auth/AuthUser.kt
package com.prototype.architecture.layer_03_domain.auth

import com.prototype.architecture.layer_03_domain.user.UserId
import com.prototype.architecture.layer_03_domain.userrole.UserRoleType
import org.springframework.security.core.Authentication
import org.springframework.security.core.GrantedAuthority

data class AuthUser(
        var userId: UserId,
        var userRoleTypes: List<UserRoleType>
) : Authentication {

    override fun getAuthorities(): Collection<GrantedAuthority?> {
        return userRoleTypes
    }

    override fun getCredentials(): Any {
        return ""
    }

    override fun getDetails(): Any {
        return ""
    }

    override fun getPrincipal(): Any {
        return userId
    }

    override fun isAuthenticated(): Boolean {
        return true
    }

    @Throws(IllegalArgumentException::class)
    override fun setAuthenticated(isAuthenticated: Boolean) {
    }

    override fun getName(): String {
        return ""
    }
}